SPEC-draft.txt

```diff
@@ -0,0 +1,267 @@
+   decisions must not be made that make this impossible.
+
+   3.3.  Performance
+
+   Performance is critical for a lot of applications. So hosting
+   runtimes SHOULD automatically optimise as much as possible. And
+   both framework and application developers SHOULD be able to 
+   easily customise the behaviour of the server component to improve the
+   performance of the application.
+
+   3.4.  Developer Experience
+
+   No software ecosystem thrives without a focus on the experience of
+   the developers using it. We recognise 3 different types of
+   developers, each with different skills and objectives.
+
+   3.4.1.  Application developers
+
+   Application developers are the developers working on the frontend
+   application that will be used by end users.
+   
+   The experience they have with FABs are the most important one to
+   consider. Both because there are many more of them than both
+   framework and hosting runtime developers, but also because they have
+   less experience and knowledge when it comes to writing reliable and
+   performant applications to be able to make the right trade-off.
+
+   A lot of these developers may not even know they are using FABs.
+
+   3.4.2.  Application framework developers
+
+   Application framework developers are the people working on the
+   building blocks that are used by the Application Developers to
+   create their applications. It is these application frameworks that
+   may have logic to output a FAB as a deployment artefact.
+
+   By making it easy to output a FAB it will lower the bar for different
+   frameworks to support it natively.
+
+   3.4.3.  Hosting Provider developers
+
+   These are the developers working on the hosting runtimes. Their
+   experience isn't about making it easier, but to allow them to build
+   the best possible runtime that their hosting platform will allow
+   them to build.
+
+
+4.  Frontend Application Bundle format
+
+   A FAB is a regular zip file which mirrors the two components of a 
+   frontend application. It includes a director called '_assets', which
+   includes all the static assets that are used by the application and
+   a file called 'server.js' that handles all server logic.
+
+   -- server.js
+    |
+    - _assets - main.abcd1234.js
+              |
+              - main.bcdef567.css
+
+   4.1  Static Assets
+   
+   The '_assets' directory contains all static assets for the
+   application such as client side Javascript and CSS files, images,
+   configuration files or templates or any other content isn't
+   dynamically generated.
+
+   Because these files will be cached for a very long time in Content
+   Delivery Networks, intermediate proxies and client browsers, it is
+   RECOMMENDED that the filename includes a unique fingerprint based on
+   the content of the file such as (a part of) a hash of the contents.
+   
+   4.1.1  URL pathname
+
+   Hosting runtimes MUST make the files available on the '/_assets/' 
```

Comment:
This doesn't seem as required as the previous point?

Reply:
Yeah, per our conversation last night I will probably lessen this requirement, so that hosting runtimes can be configured to not serve those assets..